packages/experiments/src/components/Microfeedback/Microfeedback.base.tsx

```diff
@@ -0,0 +1,161 @@
+      vote: 'no_vote',
+      isFollowupVisible: false
+    };
+  }
+
+  public render() {
+    const likeIcon = this.state.vote === 'like' ? 'LikeSolid' : 'Like';
+    const dislikeIcon = this.state.vote === 'dislike' ? 'DislikeSolid' : 'Dislike';
+    const hideThumbsDownCallout = this.state.vote !== 'dislike' || !this.state.isFollowupVisible;
+    const hideThumbsUpCallout = this.state.vote !== 'like' || !this.state.isFollowupVisible;
+
+    const classNames = getClassNames(this.props.styles, {
+      theme: this.props.theme
+    });
+
+    const likeVote = () => {
+      this._vote('like');
+    };
+    const dislikeVote = () => {
+      this._vote('dislike');
+    };
+
+    return (
+      <Stack className={classNames.root} horizontal styles={microfeedbackStyles}>
+        <div ref={this.likeRef}>
+          <IconButton menuIconProps={{ iconName: likeIcon }} title={this.props.thumbsUpTitle} onClick={likeVote} />
+        </div>
+        <div ref={this.dislikeRef}>
+          <IconButton menuIconProps={{ iconName: dislikeIcon }} title={this.props.thumbsDownTitle} onClick={dislikeVote} />
+        </div>
+        {this.props.thumbsUpQuestion ? (
+          <Callout
+            hidden={hideThumbsUpCallout}
+            role="alertdialog"
+            gapSpace={0}
+            target={this.likeRef.current}
+            setInitialFocus={true}
+            onDismiss={this._onCalloutDismiss}
+          >
+            <FocusZone direction={FocusZoneDirection.vertical}>
+              <Text block={true} styles={microfeedbackTextStyles} variant="small">
+                {this.props.thumbsUpQuestion.question}
+              </Text>
+              <List items={this.props.thumbsUpQuestion.options} onRenderCell={this._onRenderCalloutItem} />
+            </FocusZone>
+          </Callout>
+        ) : null}
+        {this.props.thumbsDownQuestion ? (
+          <Callout
+            hidden={hideThumbsDownCallout}
+            role="alertdialog"
+            gapSpace={0}
+            target={this.dislikeRef.current}
+            setInitialFocus={true}
+            onDismiss={this._onCalloutDismiss}
+          >
+            <FocusZone direction={FocusZoneDirection.vertical}>
+              <Text block={true} styles={microfeedbackTextStyles} variant="small">
+                {this.props.thumbsDownQuestion.question}
+              </Text>
+              <List items={this.props.thumbsDownQuestion.options} onRenderCell={this._onRenderCalloutItem} />
+            </FocusZone>
+          </Callout>
+        ) : null}
+      </Stack>
+    );
+  }
+
+  private _onCalloutDismiss = (): void => {
+    this.setState({ isFollowupVisible: false });
+  };
+
+  private _onRenderCalloutItem = (item: string, index: number | undefined): JSX.Element => {
+    const listOption = (): void => {
+      this._onCalloutDismiss();
+      if (this.props.sendFollowupIndex) {
+        this.props.sendFollowupIndex(index!);
```

Comment:
Generally it's a good idea not to use `!` non-null assertion operator because it can highlight an unaddressed condition. In this case, do you really want to call `sendFollowupIndex` if `index` is `undefined`? I think this could just only call `sendFollowupIndex` when `index` is defined.

Reply:
Fixed